client/rest-high-level/src/main/java/org/elasticsearch/client/eql/EqlSearchResponse.java

```diff
@@ -165,17 +170,27 @@ public int hashCode() {
                     return BytesReference.bytes(builder);
                 }
             }, SOURCE);
+            PARSER.declareObject(optionalConstructorArg(), (p, c) -> {
```

Comment:
See my comment regarding the previous version which the new parser should still be able to understand.
As it stands right now, the new client breaks backwards compatibility. 
We expect only us to be able to create the Event objects however since the constructor is public (due to parsing) we need to be mindful of versioning issues.

Reply:
@costin the "previous" version of the server's EqlSearchResponse was sending the `fields` construct already, if it was requested by user. See https://github.com/elastic/elasticsearch/blob/7.16/x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/action/EqlSearchResponse.java#L287. But the old version of the HLRC eql search response parser [here](https://github.com/elastic/elasticsearch/blob/7.16/client/rest-high-level/src/main/java/org/elasticsearch/client/eql/EqlSearchResponse.java#L159-L168) apparently ignored the "unknown" construct for `fields` based on [this logic](https://github.com/elastic/elasticsearch/blob/7.16/libs/x-content/src/main/java/org/elasticsearch/xcontent/ObjectParser.java#L303-L304) and the parameter used to build the parser [here](https://github.com/elastic/elasticsearch/blob/7.16/client/rest-high-level/src/main/java/org/elasticsearch/client/eql/EqlSearchResponse.java#L155).

So, the new HLRC parser will understand the `fields` construct since now it is configured to parse the previously ignored `fields` construct sent by the server.